Content.Shared/Emag/Systems/EmagSystem.cs

```diff
@@ -79,6 +80,13 @@ public bool DoEmagEffect(EntityUid user, EntityUid target)
         if (HasComp<EmaggedComponent>(target))
             return false;
 
+        var onAttemptEmagEvent = new OnAttemptEmagEvent(user);
+        RaiseLocalEvent(target, ref onAttemptEmagEvent);
+
+        // prevent emagging if attempt fails
+        if (!onAttemptEmagEvent.Handled)
+            return false;
```

Comment:
should this not be flipped? As it stands, every other emag effect will fail because nothing will handle this event and it'll return early here.

Reply:
woops, fixed it now